ext/datadog_profiling_native_extension/private_vm_api_access.c

```diff
@@ -479,6 +480,16 @@ int ddtrace_rb_profile_frames(VALUE thread, int start, int limit, frame_info *st
                 continue;
             }
```

Comment:
So in the past, the same cfp would eat all the "start" whereas now it'll skip just one? Not sure if this was ever something we relied on but just highlighting this slight change in behaviour.

Reply:
This is a really great catch!

I completely glossed over it because a) we don't use `start` and b) I remembered some discussion somewhere mentioning it was actually broken.

It's actually fixed nowadays https://bugs.ruby-lang.org/issues/14607 // https://github.com/ruby/ruby/pull/8280 . I had spotted that upstream had moved the `cfp = ...` to the for header as well, but I hadn't spotted it was to fix `start`.

I did spot that upstream also added the start skip logic to the `if (cme && cme->def->type == VM_METHOD_TYPE_CFUNC)` branch below. I'll push a commit to add this one as well, so we keep closer to upstream in case we ever want to use `start`.